pkg/image/registryclient/client_mirrored.go

```diff
@@ -216,6 +216,9 @@ func (r *blobMirroredRepository) alternates(ctx context.Context, fn func(r Repos
 		if err != nil {
 			return err
 		}
+		if len(alternates) == 0 {
+			return attemptErr
+		}
```

Comment:
This looks correct but now I want to see a unit test for it.  This is the real loop we need to have unit tests on anyway.